### PR TITLE
feat(atlas): transform plan Commit fields when auto_commit disabled

### PR DESCRIPTION
**Pull request by acamq**
## Summary

Fixes an issue where the plan file still contains `Commit: YES` instructions even when `auto_commit: false` is configured, which could confuse Atlas into committing anyway.

Related to #2198 — that PR added the `auto_commit: false` config option, but the plan file content wasn't being transformed to reflect this setting. The plan file is a stronger signal than the orchestrator reminder, so seeing `Commit: YES` in the plan could cause the model to commit regardless of the config.

## Changes

1. **Plan content transformation** (`tool-execute-before.ts`): When `auto_commit: false`, Read tool output for `.sisyphus/plans/*.md` files now transforms:
   - `Commit: YES` → `Commit: NO (user disabled auto-commits)`
   - `Commit: NO` → `Commit: NO (user disabled auto-commits)`

2. **Step 8 message** (`verification-reminders.ts`): When `auto_commit: false`, Step 8 now shows:
   ```
   **STEP 8: COMMITS DISABLED**
   
   The user has disabled auto-commits (`start_work.auto_commit: false`).
   Changes will NOT be committed automatically.
   ```

3. **Unit tests** (`tool-execute-before.test.ts`): 7 tests covering transformation logic including idempotency.

## Key Implementation Details

- Transformation happens in-memory at read time — plan files on disk are **never modified**
- Regex uses negative lookahead for idempotency (won't re-transform already-transformed content)
- Backward compatible: existing behavior preserved when `auto_commit: true`

## Test Plan

```bash
bun run typecheck  # ✅ No errors
bun test src/hooks/atlas/  # ✅ 48 pass (41 original + 7 new)
```

---
## Summary by cubic
Honor `auto_commit: false` by transforming plan “Commit” fields to “NO” when reading `.sisyphus/plans/*.md`, and show a clear “commits disabled” reminder. Also fixes read-path cleanup and cross-session collisions in `pendingFilePaths`.

- **Bug Fixes**
  - Transform plan content on Read in `tool.execute.after`: “Commit: YES/NO” → “Commit: NO (user disabled auto-commits)”; idempotent, uses robust file-path extraction, handles Windows paths, and won’t match “ReCommit”.
  - Harden plan detection with precise path boundaries to avoid false positives (e.g., `foo.sisyphus-backup/...`).
  - Key `pendingFilePaths` by `sessionID:callID` and always clean up on Read to prevent memory leaks and cross-session data leaks.
  - Show “STEP 7: COMMITS DISABLED” with instructions to inform the user no commits were made and ask how to proceed; Step 8 is now “proceed to next task”.
  - Tests cover path/word-boundary cases, idempotency, auto-commit on/off, and non-plan files.

<sup>Written for commit a187c1b77c87132d89a77bb39c9f666b086494d6. Summary will update on new commits.</sup>

### CODE REVIEW

File: src/hooks/atlas/verification-reminders.ts [hunk 1]
```diff
@@ -57,9 +57,19 @@ export function buildOrchestratorReminder(
...
 `
-    : ""
+    : `
+**STEP 7: COMMITS DISABLED**
 
-  const nextStepNumber = autoCommit ? 8 : 7
+The user has disabled auto-commits (\`start_work.auto_commit: false\`).
+Changes will NOT be committed automatically.
+
+**IMPORTANT: Inform the user now:**
```
Comment (code-yeongyu) [inline src/hooks/atlas/verification-reminders.ts:66]: These instructions now conflict with the next step. Step 7 tells the orchestrator to stop and ask the user about manual commit / re-enabling auto-commit, while Step 8 still says `PROCEED TO NEXT TASK` and `DO NOT STOP`. In Boulder flow that is a behavior change from "continue autonomously, just skip the commit" to "pause after each completed task," which looks like an unintended regression when `autoCommit` is disabled.

File: src/hooks/atlas/tool-execute-after.ts [hunk 2]
```diff
@@ -46,6 +69,32 @@ export function createToolExecuteAfterHandler(input: {
...
+        pendingFilePaths.delete(mapKey)
+      }
+      if (!filePath) {
+        filePath = extractFilePath(toolOutput.metadata)
+      }
+      // Transform plan files only when autoCommit is disabled
+      if (!autoCommit && filePath && isPlanPath(filePath)) {
+        const output = toolOutput.output
+        if (output && typeof output === "string") {
+          toolOutput.output = transformPlanCommitFields(output)
```
Comment (code-yeongyu) [inline src/hooks/atlas/tool-execute-after.ts:88]: This changes the `Read` result itself, so Atlas now sees synthetic plan content instead of the file's real contents. Because the PR intentionally keeps the file on disk unchanged, any later logic that re-reads the plan or tries to edit/match the `Commit:` line can diverge from what the model previously saw. That makes `Read` non-faithful and introduces a new inconsistency even though it fixes the signaling issue.

**Comment by code-yeongyu** [comment]
[sisyphus-bot]

PR sweep status check on `dev`.

- Review decision: APPROVED
- Mergeable state: **CONFLICTING** (merge conflict with current `dev`)

Approved but blocked on merge conflicts. Please rebase onto current `dev` and resolve. Once green, the owner can merge.

Assigning `code-yeongyu` for owner visibility on the merge decision.

**Comment by code-yeongyu** [comment]
[sisyphus-bot] Hi acamq. 🙏 Thanks for the atlas plan-commit auto-rewrite work.

Picking this back up from the 5/16 approval. The PR is medium-scope (319/8 across 6 files) and shows CONFLICTING against current `dev`. The atlas prompt + plan-handling has been touched on dev (see in-flight #2602 by Jrakru which targets the atlas plan contract).

Could you rebase against current `dev` and force-push? After rebase, please confirm how this composes with #2602 if both are intended to land. The auto-rewrite of Commit fields is a focused improvement; the rebase mostly needs to track the atlas-prompt structure shifts.
